File: app/nube_agent/lib/features/agents/models/session.dart
class Session {
  final String id;
  final String agent;
  final String prompt;
  final String result;
  final String status;
  final int durationMs;
  final double costUsd;
  final String userId;
  final DateTime createdAt;

  const Session({
    required this.id,
    this.agent = '',
    this.prompt = '',
    this.result = '',
    this.status = '',
    this.durationMs = 0,
    this.costUsd = 0,
    this.userId = '',
    required this.createdAt,
  });

  factory Session.fromJson(Map<String, dynamic> json) {
    return Session(
      id: json['id'] as String? ?? '',
      agent: json['agent'] as String? ?? '',
      prompt: json['prompt'] as String? ?? '',
      result: json['result'] as String? ?? '',
      status: json['status'] as String? ?? '',
      durationMs: json['duration_ms'] as int? ?? 0,
      costUsd: (json['cost_usd'] as num?)?.toDouble() ?? 0,
      userId: json['user_id'] as String? ?? '',
      createdAt: json['created_at'] != null
          ? DateTime.parse(json['created_at'] as String)
          : DateTime.now(),
    );
  }

  String get durationFormatted {
    final secs = durationMs / 1000;
    return '${secs.toStringAsFixed(1)}s';
  }

  String get costFormatted => '\$${costUsd.toStringAsFixed(3)}';
}

File: pkg/models/session.go
package models

import "time"

// ToolCallEntry records a single tool invocation within a session.
type ToolCallEntry struct {
	Name        string `json:"name"`                   // e.g. "rubix.query_nodes"
	DurationMS  int    `json:"duration_ms,omitempty"`
	Status      string `json:"status"`                 // "ok", "error"
	Error       string `json:"error,omitempty"`
	InputBytes  int    `json:"input_bytes,omitempty"`
	OutputBytes int    `json:"output_bytes,omitempty"`
}

// Session records a single agent run.
type Session struct {
	ID              string          `json:"id"`
	Provider        string          `json:"provider"`                    // "claude", "ollama", "openai", "anthropic", "gemini"
	Model           string          `json:"model,omitempty"`             // e.g. "claude-sonnet-4-20250514", "gemma4", "gpt-4.1"
	ClaudeSessionID string          `json:"claude_session_id,omitempty"` // Claude CLI session ID for --resume
	Agent           string          `json:"agent,omitempty"`
	Prompt          string          `json:"prompt"`
	Result          string          `json:"result,omitempty"`
	Status          string          `json:"status"`
	DurationMS      int             `json:"duration_ms"`
	CostUSD         float64         `json:"cost_usd"`
	InputTokens     int             `json:"input_tokens,omitempty"`
	OutputTokens    int             `json:"output_tokens,omitempty"`
	ToolCalls       int             `json:"tool_calls,omitempty"`
	ToolCallLog     []ToolCallEntry `json:"tool_call_log,omitempty"`
	UserID          string          `json:"user_id"`
	CreatedAt       time.Time       `json:"created_at"`
}

func (s Session) GetID() string { return s.ID }
